### PR TITLE
Fix config and prediction revamp.

### PR DESCRIPTION
**Pull request by b0661**
Extend single_test_optimization.py to be able to use real world data from new prediction classes.
- .venv/bin/python single_test_optimization.py --real_world --verbose
Can also be run with profiling "--profile".

Add single_test_prediction.py to fetch predictions from remote prediction providers
- .venv/bin/python single_test_prediction.py --verbose --provider-id PVForecastAkkudoktor | more
- .venv/bin/python single_test_prediction.py --verbose --provider-id LoadAkkudoktor | more
- .venv/bin/python single_test_prediction.py --verbose --provider-id ElecPriceAkkudoktor | more
- .venv/bin/python single_test_prediction.py --verbose --provider-id BrightSky | more
- .venv/bin/python single_test_prediction.py --verbose --provider-id ClearOutside | more
Can also be run with profiling "--profile".

single_test_optimization.py is an example on how to retrieve prediction data for optimization and use it to set up the optimization parameters.

Changes:
- load: Only one load provider at a time (vs. 5 before)

Bug fixes:
- prediction: only use providers that are enabled to retrieve or set data.
- prediction: fix pre pendulum format strings
- dataabc: Prevent error when resampling data with no datasets.

### CODE REVIEW

File: tests/test_prediction.py [hunk 1]
```diff
@@ -28,28 +26,14 @@ def sample_settings(reset_config):
...
-        "pvforecast2_inverter_paco": 2000,
-        "pvforecast3_peakpower": 1.6,
-        "pvforecast3_surface_azimuth": 5,
-        "pvforecast3_surface_tilt": 45,
-        "pvforecast3_userhorizon": [45, 25, 30, 60],
-        "pvforecast3_inverter_paco": 1400,
-        "pvforecast4_peakpower": None,
+        "weather_provider": None,
+        "pvforecast_provider": None,
+        "load0_provider": None,
```
Comment (Lasall) [inline tests/test_prediction.py:31]: load_provider

File: src/akkudoktoreos/prediction/loadakkudoktor.py [hunk 2]
```diff
@@ -39,8 +39,8 @@ def load_data(self) -> np.ndarray:
             profile_data = np.array(
                 list(zip(file_data["yearly_profiles"], file_data["yearly_profiles_std"]))
             )
-            data_year_energy = profile_data * self.config.loadakkudoktor_year_energy
-            # pprint(self.data_year_energy)
+            # Calculate values in W by relative profile data and yearly consumption given in kWh
+            data_year_energy = profile_data * self.config.loadakkudoktor_year_energy * 1000
```
[the remark this answers is not in the document]
Comment (b0661) [reply src/akkudoktoreos/prediction/loadakkudoktor.py:43]: The config loadakkudoktor_year_energy is in €/kWh. As this is what most users have on their bill. Optimization parameters are in €/Wh.  

```python
class LoadAkkudoktorCommonSettings(SettingsBaseModel):
    """Common settings for load data import from file."""

    loadakkudoktor_year_energy: Optional[float] = Field(
        default=None, description="Yearly energy consumption (kWh)."
    )
```

File: tests/test_prediction.py (hunk 1 above)
Comment (b0661) [reply tests/test_prediction.py]: Will change

File: src/akkudoktoreos/prediction/elecpriceakkudoktor.py [hunk 3]
```diff
@@ -146,6 +146,11 @@ def _update_data(self, force_update: Optional[bool] = False) -> None:
                 elecprice_marketprice=akkudoktor_data.values[i].marketpriceEurocentPerKWh,
             )
             self.append(record)
+        if len(self) == 0:
+            # Got no valid forecast data
```
[the remark this answers is not in the document]
Comment (b0661) [reply src/akkudoktoreos/prediction/elecpriceakkudoktor.py:150]: Now gets no data.

File: src/akkudoktoreos/prediction/loadakkudoktor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lasall) [reply src/akkudoktoreos/prediction/loadakkudoktor.py:43]: what's the unit of profile_data and data_year_energy?

**Review by Lasall — CHANGES_REQUESTED** [review]
Looks pretty good, only load0_provider still in src/akkudoktoreos/data/default.config.json and tests/testdata/EOS.config.json

File: src/akkudoktoreos/prediction/loadakkudoktor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (b0661) [reply src/akkudoktoreos/prediction/loadakkudoktor.py:43]: Profile data is given in portion of yearly energy consumption. @drbacke should know that better than me, I deduced that from the code.

[the remark this answers is not in the document]
Comment (Lasall) [reply src/akkudoktoreos/prediction/loadakkudoktor.py:43]: Just asking because of the added factor 1000

**Comment by b0661** [comment]
> Looks pretty good, only load0_provider still in src/akkudoktoreos/data/default.config.json and tests/testdata/EOS.config.json

Also removed these loadx configs.

File: src/akkudoktoreos/prediction/loadimport.py [hunk 4]
```diff
@@ -89,12 +58,7 @@ def provider_id(cls) -> str:
...
-        for load in self.loads():
-            attr_file_path = f"{load}_import_file_path"
-            attr_json = f"{load}_import_json"
-            import_file_path = getattr(self.config, attr_file_path)
-            if import_file_path is not None:
-                self.import_from_file(import_file_path, key_prefix=load)
-            import_json = getattr(self.config, attr_json)
-            if import_json is not None:
-                self.import_from_json(import_json, key_prefix=load)
+        if self.config.load_import_file_path is not None:
```
Comment (Lasall) [inline src/akkudoktoreos/prediction/loadimport.py:61]: with both parameters given the import is done twice?

**Pull request merged**